core/templates/pages/exploration-player-page/modals/refresher-exploration-confirmation-modal.component.spec.ts

```diff
@@ -74,12 +74,15 @@ describe('Refresher Exploration Confirmation Modal', () => {
     fixture = TestBed.createComponent(RefresherExplorationConfirmationModal);
     componentInstance = fixture.componentInstance;
     componentInstance.refresherExplorationId = 'test_id';
-    mockUrlService = (TestBed.inject(UrlService) as unknown) as
-      jasmine.SpyObj<MockUrlService>;
-    windowRef = (TestBed.inject(WindowRef) as unknown) as
-      jasmine.SpyObj<WindowRef>;
-    ngbActiveModal = (TestBed.inject(NgbActiveModal) as unknown) as
-      jasmine.SpyObj<NgbActiveModal>;
+    // This throws "Type object is not assignable to type
+    // 'UrlService'." We need to suppress this error
+    // because of the need to test validations. This is
+    // because the UrlService is a native class and we
+    // need to mock it.
```

Comment:
I don't understand why we cannot use the correct type

Reply:
Done